src/Line.cpp

```diff
@@ -1,127 +1,71 @@
-//
-// Created by rolf on 18-4-19.
-//
-
+#include "Print.h"
 #include "../include/roboteam_utils/Line.h"
 #include "../include/roboteam_utils/LineSegment.h"
 
 namespace rtt {
-Line::Line(const LineSegment &other) noexcept {
-    start=other.start;
-    end=other.end;
-}
-double Line::length() const {
-    return (end - start).length();
-}
-
-double Line::length2() const {
-    return (end - start).length2();
-}
-
-double Line::slope() const {
-    return (end.y - start.y)/(end.x - start.x);
-}
-
-bool Line::isVertical() const {
-    return (end.x == start.x) && (end.y != start.y);
-}
-
-Vector2 Line::direction() const {
-    return Vector2(end - start);
-}
-
-double Line::intercept() const {
-    return start.y - this->slope()*start.x;
-}
-
-std::pair<double, double> Line::coefficients() const {
-    return {slope(), intercept()};
-}
-
-bool Line::isPoint() const {
-    return start == end;
-}
-
-bool Line::isParallel(const Line &line) const {
-    // check if line is vertical, otherwise check the slope
-    if (this->isVertical() || line.isVertical()) {
-        return this->isVertical() && line.isVertical();
+Line::Line(const Vector2 &v1, const Vector2 &v2) {
+    this->v1 = v1;
+    this->v2 = v2;
+    if (v1 == v2) {
```

Comment:
This is runtime overhead i don't want in release mode.

Reply:
We should check for exceptions